packages/eslint-config-custom/react.js

```diff
@@ -0,0 +1,93 @@
+module.exports = {
+  root: true,
+  env: {
+    browser: true,
+    es2021: true,
+    node: true,
+  },
+  settings: {
+    react: {
+      version: 'detect',
+    },
+  },
+  ignorePatterns: ['dist', '.eslintrc.cjs', 'prettier.config.cjs'],
+  extends: [
+    'eslint:recommended',
+    'plugin:@typescript-eslint/recommended',
+    'plugin:react/recommended',
+    'plugin:react/jsx-runtime',
+    'plugin:react-hooks/recommended',
+  ],
+  plugins: ["unicorn"],
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    project: ['./tsconfig.json', './tsconfig.node.json'],
+  },
+  rules: {
+    'no-console': 2,
+    eqeqeq: 2,
+    "no-unused-vars": 2,
+    "no-var": 2,
+    "prefer-const": 2,
+    "no-eval": 2,
+    "no-magic-numbers": 2,
+    "unicorn/filename-case": [2, { case: "kebabCase" }],
+    "react/jsx-filename-extension": [2, { extensions: [".tsx"] }],
```

Comment:
I think we should add the following rule to disable when not import React in the component file
```
'react/react-in-jsx-scope': "off",
```

Reply:
In addition, we should add React Compiler's ESLint plugin to prepare our code to adapt to React Compiler in the future
https://react.dev/learn/react-compiler#installing-eslint-plugin-react-compiler